### DISCUSSION TITLE
Confusion about difference of poetry install w/ and w/o `--sync`

### DISCUSSION THREAD

**Discussion by Nachtalb** [python-poetry/poetry#8411] [Q&A] [answered]
Hey guys, 

According to the docs with a simple `poetry install` you get: 
> If there is a `poetry.lock` file in the current directory, it will use the exact versions from there instead of resolving them. This ensures that everyone using the library will get the same versions of the dependencies.

And for the `--sync` it says. 
> If you want to synchronize your environment – and ensure it matches the lock file – use the `--sync` option.

For me both of them sound the basically the same, `it will use the exact versions from there instead of resolving them` & `ensure it matches the lock file`. 

I can see that maybe `--sync` hinders you from installing if no `poetry.lock` exists but other than that from the docs themselves I can't see the major difference. What else other than the version would it check in the `--sync`; file hashes and PyPI repo?

If it is e.g. the file hashes, shouldn't it be the default to prevent possible unwanted behaviour (maybe even malicious) in case the developer of a package replaces a specific version. (though that's a whole other ordeal)

**Comment by Nachtalb** [comment]
It seems `--sync` does not care if the `poetry.lock` already exists or not which adds to the confusion: 

```
❯ rm poetry.lock
❯ poetry install --sync
The currently activated Python version 3.11.3 is not supported by the project (>=3.10,<3.11).
Trying to find and use a compatible version.
Using python3 (3.10.12)
Updating dependencies
Resolving dependencies... Downloading https://files.pythonhosted.org/packages/b7/ef/deacfa4f5e1a7d95e0c6d9cc2b1e8c958bfabc... 

Package operations: 0 installs, 1 update, 32 removals

  • Removing ...
```

**Reply by radoering** [reply, marked as answer]
`--sync` removes packages that are not listed in the lockfile. (PRs improving the docs are welcome.)

**Reply by Nachtalb** [reply]
Is this the only difference?

**Reply by radoering** [reply]
I think so. By the way, the old name of the option is `--remove-untracked`.

**Reply by Nachtalb** [reply]
Thanks!
